### PR TITLE
LPD-31926 Option external reference code

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPD-31926

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionResourceTest.java [hunk 1]
```diff
@@ -6,15 +6,354 @@
...
+
+		_testPatchProductOptionWithOptionExternalReferenceCode();
+	}
+
+	@Override
+	@Test
+	public void testPostProductByExternalReferenceCodeProductOptionsPage()
+		throws Exception {
+
+		ProductOption randomMappedProduct = randomProductOption();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionResourceTest.java:97]: why this variable is called _randomMappedProduct_?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionResourceTest.java [hunk 2]
```diff
@@ -6,15 +6,354 @@
...
+	@Override
+	@Test
+	public void testGraphQLDeleteProductOption() throws Exception {
+		super.testGraphQLDeleteProductOption();
+	}
+
+	@Override
+	@Test
+	public void testPatchProductOption() throws Exception {
+		ProductOption postProductOption = _postProductIdProductOptionsPage();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionResourceTest.java:71]: not sure about the method name. at line 99 there's a method that actually return a Page obj while here it is not, can you please check?

**Review by andrea-ale-sbarra — COMMENTED** [review]
couple of comments, please check
Thank you!

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**